tests/unit/test_server.py

```diff
@@ -22,7 +22,7 @@ def _get_json(file_name):
 
 
 def test_hash_validation():
-    data_hash = "sha256=38be3341f7b03bb234534be165ce4444d52bd95e798f547cabb1622db3628caa"
+    data_hash = "sha256=7127498186b8a9b282a54b72a954151d98681416693e07ea46e3a3eb960ddb42"
```

Comment:
Consider sending the test vector verbatim, so that the hash doesn't depend on how the test client chooses to encode python data into JSON, smth like this:


```py
from pathlib import Path

def test_hash_validation():
    data_hash = "sha256=712...actual-hash-here...ddb42"
    raw_json_bytes = Path("comment_created_by_bot.json").read_bytes()

    response = client.post(
        "/",
        data=raw_json_bytes,  # Send the raw JSON bytes
        headers={
            "x-hub-signature-256": data_hash,
            "Content-Type": "application/json"
        },
    )
    assert response.status_code == 200
```

Reply:
P.S. the actual HMAC of the file as present in the github repo is neither the old nor the new hash string in the test..